src/client/Client.cc

```diff
@@ -8438,6 +8454,29 @@ int Client::_open(Inode *in, int flags, mode_t mode, Fh **fhp,
     req->head.args.open.old_size = in->size;   // for O_TRUNC
     req->set_inode(in);
     result = make_request(req, perms);
+
+    /*
+     * Clients that have delegations enabled will wait on minimal caps, just
+     * to ensure that other clients holding delegations return theirs first.
+     */
+    if (deleg_timeout && result == 0) {
```

Comment:
`deleg_timeout` may be used uninitialized here:

/ceph/teuthology-archive/pdonnell-2017-10-26_13:42:21-fs-wip-pdonnell-testing-20171025.234245-testing-basic-smithi/1776790/remote/smithi016/log/valgrind/client.0.log.gz

Reply:
Good catch. I missed the initialization of deleg_timeout. I'm testing the fix now.